WordPress/Classes/ViewRelated/Post/PostListViewController.m

```diff
@@ -0,0 +1,1319 @@
+
+    // bad login/pass combination
+    EditSiteViewController *editSiteViewController = [[EditSiteViewController alloc] initWithBlog:self.blog];
+    editSiteViewController.isCancellable = YES;
+
+    UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:editSiteViewController];
+    navController.navigationBar.translucent = NO;
+
+    if (IS_IPAD) {
+        navController.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
+        navController.modalPresentationStyle = UIModalPresentationFormSheet;
+    }
+
+    [self presentViewController:navController animated:YES completion:nil];
+}
+
+
+#pragma mark - TableView Handler Delegate Methods
+
+- (NSManagedObjectContext *)managedObjectContext
+{
+    return [[ContextManager sharedInstance] mainContext];
+}
+
+- (NSFetchRequest *)fetchRequest
+{
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:NSStringFromClass([Post class])];
+    fetchRequest.predicate = [self predicateForFetchRequest];
+    NSSortDescriptor *sortDescriptorDate = [NSSortDescriptor sortDescriptorWithKey:@"date_created_gmt" ascending:NO];
+    fetchRequest.sortDescriptors = @[sortDescriptorDate];
+    fetchRequest.fetchBatchSize = PostsFetchRequestBatchSize;
+    return fetchRequest;
+}
+
+- (void)updateAndPerformFetchRequestClearingCachedRowHeights:(BOOL)clearCachedRowHeights
+{
+    NSAssert([NSThread isMainThread], @"PostsViewController Error: NSFetchedResultsController accessed in BG");
+
+    NSPredicate *predicate = [self predicateForFetchRequest];
+    NSError *error = nil;
+    [self.tableViewHandler.resultsController.fetchRequest setPredicate:predicate];
+    [self.tableViewHandler.resultsController performFetch:&error];
+    if (error) {
+        DDLogError(@"Error fetching posts after updating the fetch request predicate: %@", error);
+    }
+    if (clearCachedRowHeights) {
+        [self.tableViewHandler clearCachedRowHeights];
+    }
+    [self.tableView reloadData];
+    [self configureNoResultsView];
+}
+
+- (NSPredicate *)predicateForFetchRequest
+{
+    NSMutableArray *predicates = [NSMutableArray array];
+
+    NSPredicate *basePredicate = [NSPredicate predicateWithFormat:@"blog = %@ && original = nil", self.blog];
+    [predicates addObject:basePredicate];
+
+    NSPredicate *filterPredicate = [self currentPostListFilter].predicateForFetchRequest;
+    [predicates addObject:filterPredicate];
+
+    if (self.blog.isMultiAuthor && ![self shouldShowPostsForEveryone] && [self.blog.account.userID integerValue] > 0) {
+        NSPredicate *authorPredicate = [NSPredicate predicateWithFormat:@"authorID = %@", self.blog.account.userID];
+        [predicates addObject:authorPredicate];
+    }
+
+    NSString *searchText = self.searchController.searchBar.text;
+    if ([searchText length] > 0) {
+        NSPredicate *searchPredicate = [NSPredicate predicateWithFormat:@"postTitle CONTAINS %@", searchText];
```

Comment:
> It might not be much trouble to also search content.

I was thinking that content might have shortcodes and html, we'd need to store a plain text version in core data and keep it updated. Not terribly complicated, but not trivial either